sources/packages/web/src/services/http/ApplicationApi.ts

```diff
@@ -20,15 +20,30 @@ import {
   EnrolmentApplicationDetailsAPIOutDTO,
   CompletedApplicationDetailsAPIOutDTO,
   ApplicationAssessmentStatusDetailsAPIOutDTO,
+  ApplicationSupplementalDataAPIOutDTO,
 } from "@/services/http/dto";
 
 export class ApplicationApi extends HttpBaseClient {
   async getApplicationData(
     applicationId: number,
-  ): Promise<ApplicationDataAPIOutDTO> {
-    return this.getCall<ApplicationDataAPIOutDTO>(
-      this.addClientRoot(`application/${applicationId}`),
-    );
+  ): Promise<ApplicationDataAPIOutDTO>;
+
+  async getApplicationData(
+    applicationId: number,
+    loadDynamicData?: boolean,
+  ): Promise<ApplicationSupplementalDataAPIOutDTO>;
+
+  async getApplicationData(
+    applicationId: number,
+    loadDynamicData?: boolean,
+  ): Promise<ApplicationDataAPIOutDTO | ApplicationSupplementalDataAPIOutDTO> {
+    let url = this.addClientRoot(`application/${applicationId}`);
+    if (loadDynamicData) {
+      url = `${url}?loadDynamicData=${loadDynamicData}`;
+    }
```

Comment:
In this case, if `loadDynamicData` is false it will not be sent in the query param as it will not go inside if loop.

Suggestion:

```if (loadDynamicData !== undefined) {
      url = `${url}?loadDynamicData=${loadDynamicData}`;
    }ts
```